core/src/main/java/org/mule/transport/polling/watermark/selector/ComparableWatermarkSelector.java

```diff
@@ -40,12 +48,11 @@ public final void acceptValue(Object value)
         }
         else if (value instanceof Comparable)
         {
-            Comparable<Object> current = (Comparable<Object>) this.value;
-
-            if (current == null
-                || ((Comparable<Object>) value).compareTo(current) == this.comparableQualifier())
```

Comment:
Because of the comments on the definition of the compare method, I would just normalize the result of ((Comparable<Object>) value).compareTo(current) to fit into the [-1; 0; 1] domain (except you have a better idea of course)